src/utils/multiStepUtils.ts

```diff
@@ -163,8 +163,10 @@ export class MultiStepInput {
     const disposables: Disposable[] = [];
     const displaySteps: boolean = typeof step !== 'undefined' && typeof totalSteps !== 'undefined';
 
-    return await new Promise<string | (P extends { buttons: (infer I)[] } ? I : never)>(async (resolve, reject) => {
-      const input: InputBox = window.createInputBox();
+    const input: InputBox = window.createInputBox();
```

Comment:
Also move the config for the input box outside of the promise constructor (I think you should be able to keep the steps in the same order)